engine/src/execution_graph/logic_controllers/CacheMachine.h

```diff
@@ -51,13 +51,15 @@ class CacheMachine {
 
 	virtual void put(size_t index, std::unique_ptr<ral::frame::BlazingTable> table);
 
+	virtual void put(size_t index, std::unique_ptr<ral::cache::CacheData> cacheData);
+
 	virtual std::unique_ptr<ral::frame::BlazingTable> get_or_wait(size_t index);
 
 	virtual std::unique_ptr<ral::cache::CacheData> get_or_wait_CacheData(size_t index);
 
 	virtual void clear();
 
-	virtual bool addToCache(std::unique_ptr<ral::frame::BlazingTable> table, std::string message_id = "", bool always_add = false, const MetadataDictionary & metadata = {}, bool include_meta = false, bool use_pinned = false );
+	virtual bool addToCache(std::unique_ptr<ral::frame::BlazingTable> table, std::string message_id = "", bool always_add = false, const MetadataDictionary & metadata = {}, bool use_pinned = false );
```

Comment:
this adds yet parameter these apis are starting to be confusing

Reply:
i removed a parameter to simplify, i did not add